python/ray/workflow/examples/comparisons/prefect/compute_fib_workflow.py

```diff
@@ -1,26 +1,25 @@
 import ray
 from ray import workflow
-import requests
 
 
-def fibonacci(n):
-    assert n > 0
-    a, b = 0, 1
-    for _ in range(n - 1):
-        a, b = b, a + b
-    return b
+@ray.remote
+class FibonacciActor:
+    cache = {}
+
+    def compute(self, n):
+        if n not in FibonacciActor.cache:
+            assert n > 0
+            a, b = 0, 1
+            for _ in range(n - 1):
+                a, b = b, a + b
+            FibonacciActor.cache[n] = b
+        return FibonacciActor.cache[n]
 
 
 @ray.remote
 def compute_large_fib(M: int, n: int = 1, fib: int = 1):
-    try:
-        next_fib = requests.post(
-            "https://nemo.api.stdlib.com/fibonacci@0.0.1/", data={"nth": n}
-        ).json()
-        assert isinstance(next_fib, int)
-    except AssertionError:
-        # TODO(suquark): The web service would fail sometimes. This is a workaround.
-        next_fib = fibonacci(n)
+    fibonacci_actor = FibonacciActor.remote()
```

Comment:
You'll want to share the same actor across all workflow tasks, by passing it as an arg to `compute_large_fib` instead of creating it each time. Otherwise, you don't get any caching.